libs/ledger-live-common/src/wallet-api/logic.ts

```diff
@@ -94,7 +101,9 @@ export function signTransactionLogic(
     return Promise.reject(new Error("Account required"));
   }
 
-  const parentAccount = getParentAccount(account, accounts);
+  const currency = tokenCurrency ? findTokenById(tokenCurrency) : null;
+  const signerAccount = currency ? makeEmptyTokenAccount(account as Account, currency) : account;
```

Comment:
We should avoid using the `as` keyword if possible
```suggestion
  const signerAccount = currency ? makeEmptyTokenAccount(account, currency) : account;
```